ydb/core/log_backend/log_backend.cpp

```diff
@@ -127,36 +170,46 @@ TAutoPtr<TLogBackend> CreateAuditLogUnifiedAgentBackend(
     return logBackend;
 }
 
+THolder<TLogBackend> MaybeWrapWithJsonEnvelope(THolder<TLogBackend> logBackend, const TString& jsonEnvelope) {
+    if (jsonEnvelope.empty() || !logBackend) {
+        return logBackend;
+    }
+
+    return MakeHolder<TLogBackendWithJsonEnvelope>(jsonEnvelope, std::move(logBackend));
+}
+
 TMap<NKikimrConfig::TAuditConfig::EFormat, TVector<THolder<TLogBackend>>> CreateAuditLogBackends(
         const TKikimrRunConfig& runConfig,
         NMonitoring::TDynamicCounterPtr counters) {
     TMap<NKikimrConfig::TAuditConfig::EFormat, TVector<THolder<TLogBackend>>> logBackends;
-    if (runConfig.AppConfig.HasAuditConfig() && runConfig.AppConfig.GetAuditConfig().HasStderrBackend()) {
-        auto logBackend = NActors::CreateStderrBackend();
-        auto format = runConfig.AppConfig.GetAuditConfig().GetStderrBackend().GetFormat();
-        logBackends[format].push_back(std::move(logBackend));
-    }
 
-    if (runConfig.AppConfig.HasAuditConfig() && runConfig.AppConfig.GetAuditConfig().HasFileBackend()) {
-        auto logBackend = CreateAuditLogFileBackend(runConfig);
-        if (logBackend) {
-            auto format = runConfig.AppConfig.GetAuditConfig().GetFileBackend().GetFormat();
+    if (runConfig.AppConfig.HasAuditConfig()) {
+        const auto& auditConfig = runConfig.AppConfig.GetAuditConfig();
+        if (auditConfig.HasStderrBackend()) {
+            auto logBackend = MaybeWrapWithJsonEnvelope(NActors::CreateStderrBackend(), auditConfig.GetStderrBackend().GetLogJsonEnvelope());
+            auto format = auditConfig.GetStderrBackend().GetFormat();
             logBackends[format].push_back(std::move(logBackend));
         }
-    }
 
-    if (runConfig.AppConfig.HasAuditConfig() && runConfig.AppConfig.GetAuditConfig().HasUnifiedAgentBackend()) {
-        auto logBackend = CreateAuditLogUnifiedAgentBackend(runConfig, counters);
-        if (logBackend) {
-            auto format = runConfig.AppConfig.GetAuditConfig().GetUnifiedAgentBackend().GetFormat();
-            logBackends[format].push_back(std::move(logBackend));
+        if (auditConfig.HasFileBackend()) {
+            auto logBackend = MaybeWrapWithJsonEnvelope(CreateAuditLogFileBackend(runConfig), auditConfig.GetFileBackend().GetLogJsonEnvelope());
+            if (logBackend) {
+                auto format = auditConfig.GetFileBackend().GetFormat();
+                logBackends[format].push_back(std::move(logBackend));
+            }
```

Comment:
`MaybeWrapWithJsonEnvelope()` stands in a way here a little. I would have moved it to the very end, to `logBackends[format].push_back(...)` line -- when the main logic is done. Also one less check to make in `MaybeWrapWithJsonEnvelope()`.